frontend/app/src/components/calendar/CalendarReminder.vue

```diff
@@ -0,0 +1,313 @@
+
+const { t } = useI18n();
+
+const showReminders: Ref<boolean> = ref(false);
+
+const temporaryData: Ref<CalendarReminderTemporaryPayload[]> = ref([]);
+
+const length = computed(() => get(temporaryData).filter(item => item.secsBefore > 0).length);
+
+const {
+  fetchCalendarReminders,
+  addCalendarReminder,
+  editCalendarReminder,
+  deleteCalendarReminder,
+} = useCalendarReminderApi();
+
+const { notify } = useNotificationsStore();
+
+async function addCalendarReminderHandler(reminders: CalenderReminderPayload[]) {
+  try {
+    const result = await addCalendarReminder(reminders);
+    if (result.failed && result.failed.length > 0) {
+      notify({
+        display: true,
+        title: t('calendar.reminder.add_error.title'),
+        message: t('calendar.reminder.add_error.some_failed', {
+          ids: result.failed.join(', '),
+        }),
+      });
+    }
+  }
+  catch (error: any) {
+    logger.error(error);
+    notify({
+      display: true,
+      title: t('calendar.reminder.add_error.title'),
+      message: t('calendar.reminder.add_error.message', {
+        message: error.message,
+      }),
+    });
+  }
+}
+
+async function refreshTemporaryData() {
+  const item = get(editableItem);
+  if (item) {
+    try {
+      const identifier = item.identifier;
+      const reminders = await fetchCalendarReminders({ identifier });
+      const sortedReminders = reminders.sort((a, b) => a.identifier - b.identifier);
+      const oldData = [...get(temporaryData)].filter(item => item.isTemporary);
+      const newData = [
+        ...sortedReminders.map(item => ({ ...item, isTemporary: false })),
+        ...oldData,
+      ];
+      set(temporaryData, newData);
+    }
+    catch (error: any) {
+      logger.error(error);
+      notify({
+        display: true,
+        title: t('calendar.reminder.fetch_error.title'),
+        message: t('calendar.reminder.fetch_error.message', {
+          message: error.message,
+        }),
+      });
+    }
+  }
+}
+
+function isSameSecsBeforeExist(seconds: number) {
+  return get(temporaryData).filter(item => !item.isTemporary).some(item => item.secsBefore === seconds);
+}
+
+// 15 minutes as default value
+async function addReminder(secsBefore: number = 900, inTimeReminder = false) {
+  if (!inTimeReminder)
+    set(showReminders, true);
+
+  const item = get(editableItem);
+
+  if (!item || isSameSecsBeforeExist(secsBefore) || inTimeReminder) {
+    const newId = Date.now();
+    const newData: CalendarReminderTemporaryPayload = {
+      secsBefore,
+      identifier: newId,
+      isTemporary: true,
+    };
+
+    set(temporaryData, [
+      ...get(temporaryData),
+      newData,
+    ]);
+
+    nextTick(() => {
+      const divWrapper = document
+        .querySelector(`#calendar-reminder-amount-input-${newId}`);
+      if (divWrapper) {
+        const input = divWrapper.getElementsByTagName('input');
+        if (input[0])
+          input[0].select();
+      }
```

Comment:
no better way to do this? which component is it? where is it?

Reply:
this is to highlight the new reminder input field.
I move the implementation to `<CalendarReminderEntry />` instead.